slam-libraries/viam-cartographer/src/slam_service/slam_service.cc

```diff
@@ -22,6 +22,37 @@ ::grpc::Status SLAMServiceImpl::GetPosition(ServerContext *context,
     LOG(ERROR) << "GetPosition is not yet implemented.\n";
     return grpc::Status(grpc::StatusCode::UNIMPLEMENTED,
                         "GetPosition is not yet implemented.");
+    cartographer::transform::Rigid3d global_pose;
+    {
+        std::lock_guard<std::mutex> lk(map_builder_mutex);
+        global_pose = latest_global_pose;
```

Comment:
[q] Once we copy `latest_global_pose` into `global_pose`, would changing `latest_global_pose` also change `global_pose`? I want to check whether or not it is safe to use `global_pose` after releasing the mutex.

Reply:
I believe it should not change, we are just assigning the values directly here

Reply:
I think for this I'd probably check what happens if you copy `Eigen::Matrix<FloatType, 3, 1>` and `Eigen::Quaternion<FloatType>`, since those are underlying the `Rigid3d` data structure from what I can see. Question is if those are safe to copy & access later without a mutex

Reply:
Verified this was safe with kat in person.